source/Engine.cpp

```diff
@@ -1564,11 +1564,30 @@ void Engine::CalculateStep()
 		DoScanning(it);
 
 	// Draw the objects. Start by figuring out where the view should be centered:
-	Point newCenter = center;
+	newCenter = center;
 	Point newCenterVelocity;
 	if(flagship)
 	{
-		newCenter = flagship->Center();
+		// Code block to calculate the center of the viewport.
+		// If Camera Acceleration is enabled an offset calculated through the
+		// flagship's velocity will be added to the center
+
+		double prefMul = Preferences::CameraAcceleration() == "on" ? 1.
+			: (Preferences::CameraAcceleration() == "reversed" ? -1. : 0.);
+		if(prefMul != 0.)
+		{
+			// If the flagship isn't thrusting, because no drag is experienced,
+			// gently slide the flagship back to the center of the screen.
+			if(flagship->IsThrusting() || flagship->IsReversing())
+				offset += (flagship->Velocity() - offset) / 64.;
+			else
+				offset -= offset.Unit() / 32.;
+
+			newCenter = !flagship->IsHyperspacing() ?
+				flagship->Center() + offset * prefMul * 8 : flagship->Center();
+		}
+		else
+			newCenter = flagship->Center();
```

Comment:
```suggestion
		// Code block to calculate the center of the viewport.
		// If Camera Acceleration is enabled an offset calculated through the
		// flagship's velocity will be added to the center.
		newCenter = flagship->Center();
		double prefMul = Preferences::CameraAcceleration() == "on" ? 1.
			: (Preferences::CameraAcceleration() == "reversed" ? -1. : 0.);
		if(prefMul != 0. && !flagship->IsHyperspacing())
		{
			// If the flagship isn't thrusting, because no drag is experienced,
			// gently slide the flagship back to the center of the screen.
			if(flagship->IsThrusting() || flagship->IsReversing())
				offset += (flagship->Velocity() - offset) / 64.;
			else
				offset -= offset.Unit() / 32.;

			newCenter += offset * prefMul * 8;
		}
```